components/google_analytics/sources/new-goal/new-goal.mjs

```diff
@@ -0,0 +1,56 @@
+import analytics from "../../google_analytics.app.mjs";
+
+const LAST_GOAL_ID = "last_goal_id";
+
+export default {
+  key: "google_analytics-new-goal",
+  name: "New Goal",
+  description: "Emit new event each time a new goal is added",
+  type: "source",
+  version: "0.0.1",
+  dedupe: "greatest",
+  props: {
+    db: "$.service.db",
+    timer: {
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 60 * 15,
+      },
+    },
+    analytics,
+  },
+  methods: {
+    _getLastGoalId() {
+      return this.db.get(LAST_GOAL_ID) || 1;
+    },
+    _setLastGoalId(id) {
+      this.db.set(LAST_GOAL_ID, id);
+    },
+  },
+  async run() {
+    const startIndex = this._getLastGoalId();
+    let params = {
+      "accountId": "~all",
+      "webPropertyId": "~all",
+      "profileId": "~all",
```

Comment:
Could we have `accountId`, `webPropertyId`, and `profileId` as props in this component? Then the user can select which ones to watch.